libs/designsystem/modal/src/modal-wrapper/modal-wrapper.component.spec.ts

```diff
@@ -648,4 +651,76 @@ describe('ModalWrapperComponent', () => {
       });
     });
   });
+
+  describe('listenForScroll', () => {
+    afterEach(() => {
+      TestHelper.resetTestWindow();
+    });
+    it('should set scrollEventsEnabled to be false when opened on desktop', async () => {
+      await TestHelper.resizeTestWindow(TestHelper.screensize.desktop);
+
+      modalWrapperTestBuilder = new ModalWrapperTestBuilder(createComponent);
+      spectator = modalWrapperTestBuilder
+        .flavor('modal')
+        .title('test')
+        .component(TitleEmbeddedComponent)
+        .build();
+
+      expect(spectator.component.scrollEventsEnabled).toBeFalsy();
+    });
+
+    it('should set scrollEventsEnabled to be true when opened on a phone', async () => {
+      await TestHelper.resizeTestWindow(TestHelper.screensize.phone);
+
+      modalWrapperTestBuilder = new ModalWrapperTestBuilder(createComponent);
+      spectator = modalWrapperTestBuilder
+        .flavor('modal')
+        .title('test')
+        .component(TitleEmbeddedComponent)
+        .build();
+
+      expect(spectator.component.scrollEventsEnabled).toBeTruthy();
+    });
+
+    it('should set scrollEventsEnabled to be true when resizing from desktop to phone', async () => {
+      await TestHelper.resizeTestWindow(TestHelper.screensize.desktop);
+      modalWrapperTestBuilder = new ModalWrapperTestBuilder(createComponent);
+      spectator = modalWrapperTestBuilder
+        .flavor('modal')
+        .title('test')
+        .component(TitleEmbeddedComponent)
+        .build();
+      expect(spectator.component.scrollEventsEnabled).toBeFalse();
+
+      await TestHelper.resizeTestWindow(TestHelper.screensize.phone);
+      await TestHelper.whenTrue(() => spectator.component.scrollEventsEnabled);
+
+      expect(spectator.component.scrollEventsEnabled).toBeTrue();
+    });
+
+    it('should set border-bottom-color on ion-toolbar when scrolling on phone to the bottom or past offset', async () => {
+      await TestHelper.resizeTestWindow(TestHelper.screensize.phone);
+      modalWrapperTestBuilder = new ModalWrapperTestBuilder(createComponent);
+      spectator = modalWrapperTestBuilder
+        .flavor('modal')
+        .component(DummyContentEmbeddedComponent)
+        .build();
+      await spectator.fixture.whenStable();
+      const ionContentElement = spectator.query('ion-content') as HTMLElement;
+      ionContentElement.style.minHeight = '500px';
+      spectator.detectChanges();
+
+      spectator.component.scrollToBottom();
+      await spectator.fixture.whenStable();
+      spectator.detectChanges();
+      await new Promise((resolve) => setTimeout(resolve, 500));
```

Comment:
Please replace with more performant state detection:
```suggestion
      await TestHelper.whenTrue(() => spectator.component.isContentScrolled);
```